zlib-rs/src/inflate/bitreader.rs

```diff
@@ -77,6 +77,11 @@ impl<'a> BitReader<'a> {
         self.end as usize - self.ptr as usize
     }
 
+    #[inline(always)]
+    pub fn bytes_remaining_including_buffer(&self) -> usize {
+        (self.end as usize - self.ptr as usize) + (self.bits_used as usize >> 3)
```

Comment:
Using `/ 8` instead of `>> 3` may be a bit clearer and will optimize to the same assembly.

Reply:
we can make a pass over the repo at some point. I think zlib-ng never updated this, and the original zlib that it forked was written at a time when apparently that optimization was not (effectively) guaranteed.